CDCgov/prime-reportstream
# Platform/angela/16191/always execute send

This PR ...

Ensures that files destined for receivers with no transport configured still get an action logged for the send step. It also logs a send warning for receivers that have a transport type that is not yet supported by the SendFunction.

Test Steps:
1. Send a file on the covid-19 topic with ordering facility county set to HL7_NULL (Or set up another receiver with a null transport and send a file that will go to that receiver)
2. Verify that the file gets a record in the Action table with action_name "send" and action_result stating that it is going to the Null Transport. (I find it easiest to call the history API after the send step has executed, search the report_file table for the report_id of the file sent to the null transport receiver, and then check the associated action_id on the action table)

## Changes
- Adds a new field to Receiver for the TransportType which will return NullTransportType if there is no TransportType configured
- Modifies the SendFunction to use the new field to determine the transport that should be used
- Modifies the send_warning on the SendFunction to be more informational in the case that a TransportType is added that is not supported by the SendFunction's Transports.

## Checklist

### Testing

## Linked Issues
- Fixes #16191
